runtime/protocol/aws-xml-protocols/common/src/aws/smithy/kotlin/runtime/awsprotocol/xml/Ec2QueryErrorDeserializer.kt

```diff
@@ -14,7 +14,7 @@ internal data class Ec2QueryErrorResponse(val errors: List<Ec2QueryError>, val r
 internal data class Ec2QueryError(val code: String?, val message: String?)
 
 @InternalApi
-public fun parseEc2QueryErrorResponse(payload: ByteArray): ErrorDetails {
+public suspend fun parseEc2QueryErrorResponse(payload: ByteArray): ErrorDetails {
```

Comment:
question: this `suspend` is now unnecessary but I'm assuming you kept it for backwards-compatibility. should we also deprecate this `suspend fun` and create a new non-suspending function? 

same for `parseRestXmlErrorResponse`

Reply:
We could I chose not to for now but don't feel strongly. Yes I kept it for binary compat, this wasn't necessary for some time I don't think since our deserializers haven't been suspend for a very long time.